components/BaseComponents.js

```diff
@@ -12,6 +12,21 @@ export const TextButton = styled.Text`
   color: ${props => props.color || Colors.primaryGreen};
 `;
 
+export const NavButton = styled.TouchableOpacity`
+  background-color: #fff;
```

Comment:
Your thinking was really good to make `NavButton` a Base Component considering how often it's reused! Considering that this button has no background in most use cases (everywhere except hamburger), you should make it take in background color as a prop and default to transparent. You can maybe do this by using an RGBA value `rgba(255, 255, 255, 0)`.